components/url_sanitizer/browser/url_sanitizer_component_installer.cc

```diff
@@ -0,0 +1,70 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/url_sanitizer/browser/url_sanitizer_component_installer.h"
+
+#include <memory>
+#include <utility>
+
+#include "base/base_paths.h"
+#include "base/bind.h"
+#include "base/command_line.h"
+#include "base/containers/flat_set.h"
+#include "base/json/json_reader.h"
+#include "base/logging.h"
+#include "base/task/thread_pool.h"
+#include "base/types/expected.h"
+#include "brave/components/brave_component_updater/browser/dat_file_util.h"
+#include "brave/components/brave_component_updater/browser/local_data_files_service.h"
+#include "net/base/registry_controlled_domains/registry_controlled_domain.h"
+
+using brave_component_updater::LocalDataFilesObserver;
+using brave_component_updater::LocalDataFilesService;
+
+namespace brave {
+
+const char kCleanURLsConfigFile[] = "clean-urls.json";
```

Comment:
So this installer is piggybacking off the `Brave Local Data Updater` component? (used for debounce list, etc)

Where does the `clean-urls.json` live? (in which repo?). Curious if an update was made, what steps would have to be taken to push new package out

Reply:
https://github.com/brave/brave-core-crx-packager/pull/464, it downloaded by `Brave Local Data Updater` component

Reply:
https://github.com/brave/adblock-lists/blob/master/brave-lists/clean-urls.json is where the list lives

Reply:
Updates are supposed to be automatic, but there's currently a problem with that (https://github.com/brave/adblock-lists/issues/955) and I need to look into it with @antonok-edm.

Reply:
This is great- thanks! 😄

Reply:
@fmarier does adding this new `clean-urls.json` close out https://github.com/brave/brave-browser/issues/10188?

Reply:
I guess this can be done in another PR, but it would be cool to remove the existing hardcoded rules (ex: https://github.com/brave/brave-core/blob/d2d6b59666afa92b26e8c207ee936aad0a1c19f1/browser/net/brave_site_hacks_network_delegate_helper.cc#L31-L72) and then use this new keyed service also (`URLSanitizerService::SanitizeURL`)

Reply:
> @fmarier does adding this new `clean-urls.json` close out [brave/brave-browser#10188](https://github.com/brave/brave-browser/issues/10188)?

No, that would be in Phase 5 of https://docs.google.com/document/d/1ea7eF3s0WTsTlmMPr8LYp5KUyNg8eK4omjizo27qvJE/edit#heading=h.zfmdqhvgwdme

Reply:
@clifton Merging these two lists is something we've considered, but there are a few trade-offs and so we settled on an iterative approach (as written in the spec above). This PR is just Phase 1.